### PR TITLE
Disable existing Plutus prototype

### PR DESCRIPTION
**Pull request by erikd**
## Description

The code base included support for the Plutus prototype, but these were officially only for testing on a test net and that if transactions showed up on mainnet containing smart contract, there would be no guarantees about them. Testing shows that there are no smart contracts in the mainnet chain.

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CBR-493

## QA Steps
Synced the whole mainnet blockchain with this version of the code.

### CODE REVIEW

File: chain/test/Test/Pos/Chain/Txp/Arbitrary.hs [hunk 1]
```diff
@@ -74,7 +74,7 @@ genTxInWitness pm = oneof
     [ PkWitness <$> arbitrary <*> genSignature pm arbitrary
       -- this can generate a redeemer script where a validator script is
       -- needed and vice-versa, but it doesn't matter
-    , ScriptWitness <$> arbitrary <*> arbitrary
+
```
Comment (intricate) [inline chain/test/Test/Pos/Chain/Txp/Arbitrary.hs]: We should probably remove this whitespace and the comment above

File: chain/src/Pos/Chain/Txp/Toil/Utxo/Functions.hs [hunk 2]
```diff
@@ -252,14 +252,9 @@ verifyKnownInputs protocolMagic VTxContext {..} resolvedInputs TxAux {..} = do
...
-                redVer = scrVersion twRedeemer
-            when (valVer /= redVer) $
-                throwError $ WitnessScriptVerMismatch valVer redVer
-            when (vtcVerifyAllIsKnown && not (isKnownScriptVersion valVer)) $
-                throwError $ WitnessUnknownScriptVer valVer
-            over _Left WitnessScriptError $
-                txScriptCheck txSigData twValidator twRedeemer
+            -- The existing blockchain should not have any witnesses of this
+            -- type, so checking should just fail using the following.
+            Left $ WitnessScriptVerMismatch (scrVersion twValidator) (scrVersion twRedeemer)
```
Comment (intricate) [inline chain/src/Pos/Chain/Txp/Toil/Utxo/Functions.hs]: :100:

**Comment by avieth** [comment]
Can we drop `plutus-prototype` from cabal/stack/nix files? That would be delightful as I wouldn't have to bring it in when building cardano-sl as a dependency for an ouroboros-network application.

File: chain/test/Test/Pos/Chain/Txp/Arbitrary.hs (hunk 1 above)
Comment (erikd) [reply chain/test/Test/Pos/Chain/Txp/Arbitrary.hs]: I alrady had that as a local clean up that I forgot to push :).

**Comment by erikd** [comment]
@avieth I will have a another attempt at removing plutus-prototype as a dependency.

**Comment by disassembler** [comment]
p=0

**Comment by disassembler** [comment]
bors p=0

**Pull request merged**